### ISSUE TITLE
Similar project for iframe integration

### ISSUE THREAD

**Issue by martenrichter** [TileDB-Inc/jupyter-iframe-commands#39]
I have just found this repo.
I have been working roughly since May on a similar project.
It is for integrating Jupyter Lite with my echalk software fails-components (see  https://github.com/fails-components/compositions or https://www.youtube.com/@fails-components) with jupyter applets. I want to use it in my theoretical physics classes.

It is not finished, but you can open and save notebooks and select applets within my application;

[Recording2024-12-27 11-49-51.webm](https://github.com/user-attachments/assets/97971e1c-2a52-4bc6-9d85-41552e29015b)

I have just made it public:
https://github.com/fails-components/jupyterfails
you can find the current rect component here ( I may move it over):
https://github.com/fails-components/app/blob/master/src/jupyteredit.jsx

But it needs a lot of cleanup, and the build scripts are not ready. The package version and structure will also change. There are many issues, and the interceptor (which should allow the transmission of an applet state to the students) is not ready yet.
It uses a lot of internal monkey patching, as some interfaces do not exist yet on jupyter lab.

I wanted to post this information so that work is not done twice.

**Comment by jtpio** [comment]
Thanks @martenrichter for sharing this!

Looking at the code of this `jupyterfails` project, it seems that the scope may be a bit broader than what `jupyter-iframe-commands` tries to tackle.

`jupyter-iframe-commands` focuses on providing a way to interact with a JupyterLab (or lab-based application like JupyterLite and Notebook) running in an IFrame, via the commands system.

Maybe `jupyterfails` can leverage `jupyter-iframe-commands` if it helps simplify a few things? Also feel free to open PRs to this repo if some things need to be improved or fixed, thanks!

**Comment by martenrichter** [comment]
Thanks, @jtpio for clarification.
So the plan for `jupyter-iframe-commands` was just to send commands and not a full integration?
Ok, then the purpose is different.
I just wanted to check, if there is not another effort ongoing, as I know at least another application (lab documentation or things like next cloud, where it would make sense).

And if there is demand I can also integrate the functionality `jupyter-iframe-commands`. (So far for my use case, I most times needed a return value, which commands are not offering) but without `comlink`.

**Comment by jtpio** [comment]
> So, was the plan for `jupyter-iframe-commands` just to send commands and not a full integration?

Yes, since commands provide a convenient way to accessing functionalities within JupyterLab, and also because new ones can be added by third-party extensions easily.
